crates/symbolicator-service/src/services/ppdb_caches.rs

```diff
@@ -142,26 +169,38 @@ impl PortablePdbCacheActor {
                 scope: request.scope.clone(),
                 purpose: ObjectPurpose::Debug,
             })
-            .await
-            .map_err(|e| Arc::new(PortablePdbCacheError::Fetching(e)))?;
+            .await else {
+            return (CacheEntry::InternalError, AllObjectCandidates::default(), ObjectFeatures::default())
+        };
 
         match meta {
             Some(handle) => {
-                self.ppdb_caches
+                match self
+                    .ppdb_caches
                     .compute_memoized(FetchPortablePdbCacheInternal {
                         request,
                         objects_actor: self.objects.clone(),
                         object_meta: handle,
-                        candidates,
+                        candidates: candidates.clone(),
```

Comment:
Improvement opportunity: We don’t need to pass these along through the `FetchPortablePdbCacheInternal` / `PortablePdbCacheFile` anymore now, as we can just use the `candidates` that we already have.

This is also more correct. As `compute_memoized` does request coalescing, there is a slim chance that we end up with `candidates` from a different request.